src/app/auth.service.ts

```diff
@@ -20,41 +26,148 @@ const getRandomString = len => 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789abcdefghijkl
 @Injectable()
 export class AuthService {
 
+  public authed = new Subject<boolean>();
   private authenticated: boolean = false;
-  private access_token: string;
-  private refresh_token: string;
-  private token_type: string;
-  private expires_in: number;
+  private oauth = {};
   private state: string;
   private stateKey ='spotify_auth_state';
-  private scope = 'user-read-email user-read-currently-playing user-modify-playback-state user-read-playback-state';
-  private redirect_uri = 'http://localhost:8080/';
+  private duetify_uri = 'http://localhost:8080/';
   private spotifyAuthUrl = 'https://accounts.spotify.com/authorize?';
   private client_id = 'c165fd4b3c454bd5b1c726c172ea2faf';
+  private scope = 'user-read-email user-read-currently-playing user-modify-playback-state user-read-playback-state playlist-modify-private playlist-read-private';
 
-  constructor(private spotify: SpotifyApiService) { }
+  constructor(
+    private http: HttpClient,
+    private spotify: SpotifyApiService,
+    private localStorage: LocalStorageService,
+    private router: Router
+  ) {
+    if (this.localStorage.keys().includes('access_token')) {
+      this.oauth = this._loadFromLocalStorage();
+      this.spotify.checkToken(this.oauth['access_token'])
+      .subscribe(
+        data => {
+          if (data['type'] === 'user') {
+            this.authenticated = true;
+          }
+        }
+      );
+    }
+  }
+
+  ngOnInit () {
+    this.spotify.getUser();
+
+  }
 
   public isAuthenticated = () => this.authenticated;
-  public getAccessToken = () => this.access_token;
-
-  public checkAuthData = (data: any) => {
-    if (data.access_token) {
-      let test = this._testAccessToken(data.access_token);
-      if (test) {
-        this._setAccessToken(data.access_token);
-        this._setRefreshToken(data.refresh_token);
-        this._setExpire(data.expires_in);
-        this._setTokenType(data.token_type);
-        this._setAuthenticated();
-        console.log('Authenticated');
+  public testAuth = (): Observable<boolean> => {
+    if (this.notExpired(this.oauth['expires'])) {
+      this._setAuthenticated();
+      this.spotify._setAuthToken(this.oauth['access_token']);
+      this.authed.next(true);
+    } else if (!this.oauth['access_token']) {
+      console.log('isAuthenticated; no access_token');
+      this.authed.next(false);
+    } else {
+      this.checkAuthData(this.oauth)
+        .subscribe(
+          data => {
+            this.authed.next(data);
+          },
+          err => {
+            console.log('there was an error with checkAuthData',err);
+          }
+        );
+    }
+    return Observable.from(this.authed);
+  }
+
+  public getAccessToken = () => this.oauth['access_token'];
+
+  public checkAuthData = (d: {}): Observable<boolean> => {
```

Comment:
The colon is for assigning a type, is this what you wanted to do? Were you assigning a default value?